src/store/models/network.ts

```diff
@@ -922,6 +935,95 @@ const networkModel: NetworkModel = {
+    const { networks } = getState();
+    const network = networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    const getNodeLightningService = (node: LightningNode): LightningService => {
+      switch (node.implementation) {
+        case 'LND':
+          return lndService;
+        case 'c-lightning':
+          return clightningService;
+        case 'eclair':
+          return eclairService;
+        default:
+          throw new Error('unknown implementation');
+      }
+    };
+
+    const balanceChannel = async (
+      channel: LightningNodeChannel,
+      localNode: LightningNode,
+      remoteNode: LightningNode,
+      satsTolerance = 150,
+    ) => {
+      if (channel.status !== 'Open') {
+        // TODO: warn about channel not opened.
+        return;
+      }
+
+      if (remoteNode === undefined || remoteNode === null) {
+        // TODO: warn about remote node being null.
+        return;
+      }
+
+      const localBalance = Number(channel.localBalance);
+      const remoteBalance = Number(channel.remoteBalance);
+      const toPay = Math.floor(Math.abs(localBalance - remoteBalance) / 2);
+
+      // If the balance difference in satoshis is too small, we ignore it.
+      if (toPay < satsTolerance) {
+        return;
+      }
+
+      // The source node pays an invoice to the target node, in order to balance the channel.
+      const src = localBalance > remoteBalance ? localNode : remoteNode;
+      const target = localBalance > remoteBalance ? remoteNode : localNode;
+
+      console.log(
+        '[AUTO BALANCE]: ',
+        'paying from ' + src.name + ' to ' + target.name + ' the amount ' + toPay,
+      );
+
+      const invoice = await getNodeLightningService(target).createInvoice(target, toPay);
+
+      await getNodeLightningService(src).payInvoice(src, invoice);
+    };
+
+    interface ChannelInfo {
+      channel: LightningNodeChannel;
+      fromNode: LightningNode;
+      toNode: LightningNode;
+    }
+
+    const lnNodes = network.nodes.lightning;
+    const channels = [] as LightningNodeChannel[];
+    const id2Node = {} as Record<string, LightningNode>;
+
+    for (const node of lnNodes) {
+      const lightningService = getNodeLightningService(node);
+      const nodeChannels = await lightningService.getChannels(node);
+      channels.push(...nodeChannels);
+      id2Node[node.name] = node;
+    }
+
+    const links = getStoreState().designer.activeChart.links;
+    const channelsInfo = [] as ChannelInfo[];
+
+    for (const channel of channels) {
+      const id = channel.uniqueId;
+      const { to, from } = links[id];
+      const fromNode = id2Node[from.nodeId as string];
+      const toNode = id2Node[to.nodeId as string];
+      channelsInfo.push({ channel, fromNode, toNode });
+    }
+
+    for (const { channel, fromNode, toNode } of channelsInfo) {
+      await balanceChannel(channel, fromNode, toNode);
+    }
+  }),
```

Comment:
At the end of this function, you should call `syncChart` so that the channels display the updated balances.

```ts
await getStoreActions().designer.syncChart(network);
```